src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs

```diff
@@ -147,6 +147,9 @@ internal TypeSymbolWithAnnotations SubstituteTypeWithTupleUnification(TypeSymbol
                 var possiblyTuple = TupleTypeSymbol.TransformToTupleIfCompatible(type);
                 if ((object)type != possiblyTuple)
                 {
+                    // PROTOTYPE(NullableReferenceTypes): This ignores the particular TypeSymbolWithAnnotations
```

Comment:
We also might want to avoid resolving type eagerly several lines above ```var type = result.TypeSymbol```. Please add a comment for that as well.